src/main/java/org/ebyhr/trino/storage/operator/OrcPlugin.java

```diff
@@ -51,27 +51,56 @@
     @Override
     public List<StorageColumn> getFields(String path, Function<String, InputStream> streamProvider)
     {
-        path = getLocalPath(path, streamProvider);
-        OrcReader reader = getReader(new File(path));
-        ColumnMetadata<OrcType> types = reader.getFooter().getTypes();
-        return reader.getRootColumn().getNestedColumns().stream()
-                .map(orcColumn -> new StorageColumn(
-                        orcColumn.getColumnName(),
-                        fromOrcType(types.get(orcColumn.getColumnId()), types)))
-                .collect(Collectors.toList());
+        AutoDeletingTempFile file = null;
```

Comment:
If `getLocalPath` would always return a closeable that has a `getPath`, `try-with-resources` could be used here.

Reply:
I was thinking about something like this: #158